File: sample-app-hd-calculator/src/main/java/com/hd/calculator/app/function/store/StoreManager.java
package com.hd.calculator.app.function.store;

import com.hd.calculator.app.MyApp;
import com.tencent.mmkv.MMKV;

public class StoreManager {

    private MMKV mmkv;

    public static StoreManager getIns() {
        return StoreManagerHolder.instance;
    }

    public MMKV getDefaultMMKV() {
        return mmkv;
    }

    public void init() {
        MMKV.initialize(MyApp.context);
        mmkv = MMKV.defaultMMKV();
    }

    public void putString(String key, String value) {
        mmkv.putString(key, value);
    }

    public String getString(String key, String defaultValue) {
        return mmkv.getString(key, defaultValue);
    }

    public void putInt(String key, int value) {
        mmkv.putInt(key, value);
    }

    public int getInt(String key, int defaultValue) {
        return mmkv.getInt(key, defaultValue);
    }

    public void putLong(String key, long value) {
        mmkv.putLong(key, value);
    }

    public long getLong(String key) {
        return mmkv.getLong(key, 0);
    }

    public void putFloat(String key, float value) {
        mmkv.putFloat(key, value);
    }

    public float getFloat(String key) {
        return mmkv.getFloat(key, 0);
    }

    public void putBoolean(String key, boolean value) {
        mmkv.putBoolean(key, value);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return mmkv.getBoolean(key, defaultValue);
    }

    private static class StoreManagerHolder {
        private final static StoreManager instance = new StoreManager();
    }
}

File: sample-app-text-card/src/main/java/com/allever/android/card/text/pic/text/util/StoreManager.kt
package com.allever.android.card.text.pic.text.util

import app.allever.android.lib.core.store.StoreCore

object StoreManager {

    fun putString(key: String, value: String) {
        StoreCore.putString(key, value)
    }

    fun getString(key: String, default: String = ""): String {
        return StoreCore.getString(key, default)?:""
    }

    fun putBoolean(key: String, value: Boolean) {
        StoreCore.putBoolean(key, value)
    }

    fun getBoolean(key: String, default: Boolean = false): Boolean {
        return StoreCore.getBoolean(key, default)
    }

    fun putInt(key: String, value: Int) {
        StoreCore.putInt(key, value)
    }

    fun getInt(key: String, default: Int = 0): Int {
        return StoreCore.getInt(key, default)
    }
}